handlers/sql_generic_v3/handler/init.rb

```diff
@@ -0,0 +1,221 @@
+  space_form = @info_values["space_form"]
+  db_server = @info_values["database_server"]
+  db_port = @info_values["database_port"]
+  db_name = @parameters["dbname"]
+  db_username = @info_values['database_username']
+  db_password = @info_values["database_password"]
+  error_message = nil
+  
+
+    #checking to see if the server and port are open
+    #this was put in to test java vm access to my local sql server
+    begin
+      kinetic_api_route = "#{@info_values["kinetic_api_location"]}/datastore/forms/" +
+        "#{space_form}/submissions?index=values[Template Name]" +
+        "&q=values[Template Name]=\"#{template_name}\"&include=values"
+      puts "Calling Kinetic Platform at: #{kinetic_api_route}" if @enable_debug_logging
+      
+      response = RestClient::Request.execute \
+        method: 'GET', \
+        url: kinetic_api_route, \
+        user: @info_values["kinetic_api_username"], \
+        password: @info_values["kinetic_api_password"], \
+        headers: {:content_type => @content_type, :accept => @accept}
+      response_code = response.code
+
+      # The query should be indexed to a unique field.
+      submission = JSON.parse(response)['submissions'][0]
+      if (submission.nil?) 
+        raise "No submissions found with a Template Name of '#{template_name}'"
+      end
+
+      query_template = submission["values"]["SQL Query"]
+      if (query_template.nil? || query_template.strip.empty?) 
+        raise "The query was nil or empty.  Check that the '#{space_form}' form has the SQL Query field."
+      end
+
+      socket = TCPSocket.new(db_server, db_port)
+      puts "Port #{db_port} at server '#{db_server}' is open." if @enable_debug_logging
+      begin
+        if @parameters["jdbc_database"].downcase == "sqlserver"
+            @db = Sequel.connect("jdbc:#{@parameters["jdbc_database"]}://#{db_server}:#{db_port};database=#{db_name};user=#{db_username};password=#{db_password}")
+            @db.extension :identifier_mangling
+            @db.identifier_input_method = nil
+            @db.identifier_output_method = nil
+            @max_db_identifier_size = 128
+        elsif @parameters["jdbc_database"].downcase == "oracle"
+          Sequel.database_timezone = :utc
+            #Sequel.application_timezone = :utc
+            @db = Sequel.connect("jdbc:#{@parameters["jdbc_database"]}:thin:#{db_username}/#{db_password}@#{db_server}:#{db_port}:#{db_name}")
+            @db.extension :identifier_mangling
+            @db.identifier_input_method = nil
+            @db.identifier_output_method = nil
+            @max_db_identifier_size = 30
+        elsif @parameters["jdbc_database"].downcase == "postgresql"
+            @max_db_identifier_size = 64
+            @db = Sequel.connect("jdbc:#{@parameters["jdbc_database"]}://#{db_server}:#{db_port}/#{db_name}?user=#{db_username}&password=#{db_password}")
+        elsif @parameters["jdbc_database"].downcase == "mysql"

```

Comment:
Not a supported db, remove this. You could just add an "else" and raise exception that the chosen database is not supported. But should never get to that point if parameter is changed to use a menu with the supported databases.